docs(grid): Revamp SeachBox article (#2243)
telerik/blazor-docs: components/grid/filter/searchbox.md

        }
    }

    protected override void OnInitialized()
    {
        for (int i = 1; i <= 500; i++)
        {
            GridData.Add(new SampleModel()
            {
                Id = i,
                Name = $"{(char)Random.Shared.Next(65, 91)}{(char)Random.Shared.Next(65, 91)} " +
                    $"{(char)Random.Shared.Next(65, 91)}{(char)Random.Shared.Next(65, 91)} {i}",
                Description = $"{(char)Random.Shared.Next(97, 123)}{(char)Random.Shared.Next(97, 123)} " +
                    $"{(char)Random.Shared.Next(97, 123)}{(char)Random.Shared.Next(97, 123)} {i}"
            });
        }
    }

    public class SampleModel
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
    }
}
````

@[template](/_contentTemplates/grid/state.md#initial-state)

## Customize the SearchBox

The `GridSearchBox` component offers the following parameters to customize its behavior:

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

| Parameter | Type and Default&nbsp;Value | Description |
| --- | --- | --- |

> | `Class` | `string`| The custom CSS class on the SearchBox wrapper (`<span class="k-searchbox">`). |

Reviewer: ```suggestion
| `Class` | `string`| The custom CSS class that renders on the SearchBox wrapper (`<span class="k-searchbox">`). |
```

> | `DebounceDelay` | `int` <br /> (`300`) | The time in milliseconds when the searching starts after the user has finished typing. This provides a performance optimization when using the [`OnRead` event]({%slug common-features-data-binding-onread%}). Filtering does not occur on every keystroke during fast typing, unless `DebounceDelay` is set to `0`. |

Reviewer: ```suggestion
| `DebounceDelay` | `int` <br /> (`300`) | The time in milliseconds between the user finishing typing and the search starting. This provides a performance optimization when using the [`OnRead` event]({%slug common-features-data-binding-onread%}). Filtering does not occur on every keystroke during fast typing, unless `DebounceDelay` is set to `0`. |
```
